data/database/migration_rollback.py

```
@@ -0,0 +1,610 @@
+                raise ValueError(f"Backup integrity check failed: {result[0]}")
+            
+            # Size comparison if original provided
+            if original_path and os.path.exists(original_path):
+                original_size = os.path.getsize(original_path)
+                size_difference_pct = abs(backup_size - original_size) / original_size * 100
+                
+                if size_difference_pct > 10:  # Allow 10% difference
+                    self.logger.warning(f"Backup size differs significantly from original: {size_difference_pct:.1f}%")
+            
+        except Exception as e:
+            raise ValueError(f"Backup integrity verification failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)